frontend/appflowy_flutter/lib/workspace/presentation/home/menu/sidebar/sidebar_workspace.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class SidebarWorkspace extends StatelessWidget {
+  const SidebarWorkspace({
+    super.key,
+    required this.userProfile,
+    required this.views,
+  });
+
+  final UserProfilePB userProfile;
+  final List<ViewPB> views;
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider<UserWorkspaceBloc>(
+      create: (_) => UserWorkspaceBloc(userProfile: userProfile)
+        ..add(const UserWorkspaceEvent.fetchWorkspaces()),
+      child: BlocConsumer<UserWorkspaceBloc, UserWorkspaceState>(
+        listener: _showResultDialog,
+        builder: (context, state) {
+          final currentWorkspace = state.currentWorkspace;
+          // todo: show something if there is no workspace
+          if (currentWorkspace == null) {
+            return const SizedBox.shrink();
+          }
+          return Row(
+            children: [
+              Expanded(
+                child: _WorkspaceWrapper(
+                  userProfile: userProfile,
+                  currentWorkspace: currentWorkspace,
+                ),
+              ),
+              UserSettingButton(userProfile: userProfile),
+              const HSpace(4),
+              NotificationButton(views: views),
+            ],
+          );
+        },
+      ),
+    );
+  }
+
+  void _showResultDialog(BuildContext context, UserWorkspaceState state) {
+    var result = state.createWorkspaceResult;
+
+    if (result != null) {
+      final message = result.fold(
+        (s) => LocaleKeys.workspace_createSuccess.tr(),
+        (e) => '${LocaleKeys.workspace_createFailed.tr()}: ${e.msg}',
+      );
+      showSnackBarMessage(context, message);
+      return;
+    }
+
+    result = state.deleteWorkspaceResult;
+    if (result != null) {
+      final message = result.fold(
+        (s) => LocaleKeys.workspace_deleteSuccess.tr(),
+        (e) => '${LocaleKeys.workspace_deleteFailed.tr()}: ${e.msg}',
+      );
+      showSnackBarMessage(context, message);
+      return;
+    }
+
+    result = state.openWorkspaceResult;
+    if (result != null) {
+      final message = result.fold(
+        (s) => LocaleKeys.workspace_openSuccess.tr(),
+        (e) => '${LocaleKeys.workspace_openFailed.tr()}: ${e.msg}',
+      );
+      showSnackBarMessage(context, message);
+      return;
+    }
+  }
+}
+
+class _WorkspaceWrapper extends StatefulWidget {
+  const _WorkspaceWrapper({
+    required this.userProfile,
+    required this.currentWorkspace,
+  });
+
+  final UserWorkspacePB currentWorkspace;
+  final UserProfilePB userProfile;
+
+  @override
+  State<_WorkspaceWrapper> createState() => _WorkspaceWrapperState();
+}
+
+class _WorkspaceWrapperState extends State<_WorkspaceWrapper> {
+  final controller = PopoverController();
+
+  @override
+  Widget build(BuildContext context) {
+    if (PlatformExtension.isDesktopOrWeb) {
+      return AppFlowyPopover(
```

Comment:
Extract into eg. `DesktopWorkspaceWrapper` so it's easier to read the if-else statement